### ISSUE TITLE
UAMD-Net: A Unified Adaptive Multimodal Neural Network for Dense Depth Completion

### ISSUE THREAD

**Issue by jhonnye0** [lucasmmassa/digital_image_processing#6]
[UAMD-Net: A Unified Adaptive Multimodal Neural Network for Dense Depth
Completion](https://arxiv.org/pdf/2204.07791.pdf)

**Comment by jhonnye0** [comment]
![image](https://user-images.githubusercontent.com/43444151/164987811-616fa579-61d8-45da-aff4-4bc971f469bc.png)

**Comment by jhonnye0** [comment]
Depth prediction is a critical problem in robotics applications, especially autonomous driving. However, the former usually suffers from overfitting while building cost volume, and the latter has a limited generalization due to the lack of geometric constraint. To solve these problems, we propose a novel multimodal neural network, namely UAMD-Net, for dense depth completion based on the fusion of binocular stereo matching and the weak constrain from the sparse point clouds. Specifically, the sparse point clouds are converted to sparse depth map and sent to the multimodal feature encoder (MFE) with binocular image, constructing a cross-modal cost volume. Then, it will be further processed by the multimodal feature aggregator (MFA) and the depth regression layer. Furthermore, the existing multimodal methods ignore the problem of modal dependence, that is, the network will not work when a certain modal input has a problem. Therefore, we propose a new training strategy called Modal-dropout which enables the network to be adaptively trained with multiple modal inputs and inference with specific modal inputs. Comprehensive experiments conducted on the KITTI depth completion benchmark demonstrate that our method produces robust results and outperforms other state-of-the-art methods.

**Comment by jhonnye0** [comment]
![image](https://user-images.githubusercontent.com/43444151/164996558-e5e38e1b-efb9-4c0c-9b95-b84fefd784c6.png)
